modin/data_management/data_manager.py

```diff
@@ -1111,8 +1111,23 @@ def all(self, **kwargs):
             Pandas Series containing boolean values.
         """
         axis = kwargs.get("axis", 0)
-        func = self._prepare_method(pandas.DataFrame.all, **kwargs)
-        return self.full_axis_reduce(func, axis)
+        bool_only = kwargs.get("bool_only", None)
+        index = self.index if axis else self.columns
+        if bool_only:
+            not_bool = []
+            for index, dtype in zip(index, self.dtypes):
+                if dtype != bool:
+                    not_bool.append(index)
+
```

Comment:
please remove excess vertical whitespace.

also, inline docs would help the maintainability of the code.

do you think this part of `any` and `all` could be combined into one helper method? i'd hate to have copied code between them if we can avoid it.

Reply:
Great idea! I'm not sure about the name for the new helper method but I'll push something and let's go from there.